### ISSUE TITLE
examples/hostpath-provisioner: make does not always rebuild

### ISSUE THREAD

**Issue by pohly** [kubernetes-sigs/sig-storage-lib-external-provisioner#80]
Invoking "make" multiple times in examples/hostpath-provisioner does not rebuild the binary even if the source code has changed.

**Comment by niranjandarshann** [comment]
Yes, it's true. If invoking make multiple times in examples/hostpath-provisioner does not rebuild the binary even when the source code has changed, the issue may be  due to outdated or incorrect dependencies in the Makefile what i think.

**Comment by niranjandarshann** [comment]
I think adding  build target in .PHONY  to ensure source changes trigger a rebuild
```
.PHONY: build
build: dep
	go build -o hostpath-provisioner ./controller ./allocator

```

**Comment by niranjandarshann** [comment]
And also 
 the binary rebuilds when source code changes
```
hostpath-provisioner: $(shell find . -name '*.go')
	go build -o hostpath-provisioner ./controller ./allocator
```

**Comment by pohly** [comment]
I like marking targets as PHONY better than trying to figure out what the dependencies are.

**Comment by niranjandarshann** [comment]
got you @pohly . i think we should mark all the targets as PHONY

`.PHONY: all clean
`

Adding this to makefile . WDYT! if it look better i would like to work on it
